backend/src/manager/mod.rs

```diff
@@ -805,17 +916,57 @@ async fn get_running_ip(
             }) => (),
             Err(e) => return GetRunninIp::Error(e.into()),
         }
-        match futures::poll!(&mut runtime) {
-            Poll::Ready(res) => match res {
+        if let Poll::Ready(res) = futures::poll!(&mut runtime) {
+            match res {
                 Ok(Ok(response)) => return GetRunninIp::EarlyExit(response),
                 Err(_) | Ok(Err(_)) => {
                     return GetRunninIp::Error(Error::new(
                         eyre!("Manager runtime panicked!"),
                         crate::ErrorKind::Docker,
                     ))
                 }
-            },
-            _ => (),
+            }
+        }
+    }
+}
+
+async fn get_long_running_ip(
```

Comment:
:notebook: Just extracted logic that is shared between long running and regular docker exec